client-reactjs/src/components/application/IndexDetails.js

```diff
@@ -76,7 +110,15 @@ class IndexDetails extends PureComponent {
             id: data.basic.id,
             groupId: data.basic.groupId,
             keyedColumns: data.basic.index_keys,
-            nonKeyedColumns: data.basic.index_payloads
+            nonKeyedColumns: data.basic.index_payloads,
+
+            // For read only view
+            title: data.basic.title == '' ? data.basic.name : data.basic.title,
+            name: (data.basic.name == '' ? data.basic.title : data.basic.name),
```

Comment:
Can you check if populating these state variables can be avoided? I think antd form has a setFieldsValue() method that can be used to populate form fields, if thats what these state variables are used for. It should be already used in other functions in existing code

Reply:
We can avoid most of them, I am going to refactor. This has to be done for Markdown , checkboxes , radio buttons etc.